### PR TITLE
Fix bug 836587: Add "Source code for this page" link to footer

### PR DESCRIPTION
**Pull request by pmac**
It will be a link to the current template source on github.

Original inspiration for this commit from:

https://github.com/l-hedgehog/bedrock/commit/36f011c0

### CODE REVIEW

**Comment by pmac** [comment]
We'll obviously have to block for l10n for the new string (unless someone has ideas). But this seems pretty straight forward.

/cc @l-hedgehog @alexgibson @jpetto

**Comment by pmac** [comment]
Thoughts anyone? Shall I put up a demo? That seems like a good idea regardless. I'll link a demo here shortly.

**Comment by pmac** [comment]
Okay. You can now see this in action on [my demo](https://bedrock-demo-pmac.us-west.moz.works/en-US/). There is an "edit this page" link in the footer of most pages there.

**Comment by alexgibson** [comment]
I think this looks great, nice work @pmac!

File: bedrock/base/templates/includes/site-footer.html [hunk 1]
```diff
@@ -17,6 +17,9 @@
           <li><a href="{{ url('mozorg.contact.contact-landing') }}" data-link-type="footer" data-link-name="Contact Us">{{ _('Contact Us') }}</a></li>
           <li><a href="{{ donate_url('footer') }}" class="donate" data-link-type="footer" data-link-name="Donate">{{ _('Donate') }}</a></li>
           <li><a href="https://wiki.mozilla.org/Webdev/GetInvolved/mozilla.org" data-link-type="footer" data-link-name="Contribute to this site">{{ _('Contribute to this site') }}</a></li>
+          {% if template_source_url %}
+            <li><a href="{{ template_source_url }}" data-link-type="footer" data-link-name="Edit this page">{{ _('Edit this page') }}</a></li>
```
Comment (alexgibson) [inline bedrock/base/templates/includes/site-footer.html]: This adds a new string to the shared footer, so we'll either need to show it for `en` only temporarily, or let this sit while the new string is translated?

**Comment by jpetto** [comment]
Agree that this is a great addition. ⭐️

File: bedrock/base/templates/includes/site-footer.html [hunk 2]
```diff
@@ -17,6 +17,9 @@
           <li><a href="{{ url('mozorg.contact.contact-landing') }}" data-link-type="footer" data-link-name="Contact Us">{{ _('Contact Us') }}</a></li>
           <li><a href="{{ donate_url('footer') }}" class="donate" data-link-type="footer" data-link-name="Donate">{{ _('Donate') }}</a></li>
           <li><a href="https://wiki.mozilla.org/Webdev/GetInvolved/mozilla.org" data-link-type="footer" data-link-name="Contribute to this site">{{ _('Contribute to this site') }}</a></li>
+          {% if template_source_url %}
+            <li><a href="{{ template_source_url }}" data-link-type="footer" data-link-name="Edit this page">{{ _('Source code for this page') }}</a></li>
```
Comment (alexgibson) [inline bedrock/base/templates/includes/site-footer.html]: Should we add `rel="nofollow"` to this link, probably useful only for humans?

**Comment by pmac** [comment]
After discussing with @alexgibson on IRC I've changed the link wording to "Source code for this page". That's more accurate as well as possibly less confusing if someone has never seen Github before.

**Comment by pmac** [comment]
Okay. I think we all approve of the change and the new string. 

@peiying2 this PR contains a single new string, but it is in the footer and therefore needs to go into `main.lang`. You may get started on that when ready. Please let me know when it's well-enough translated to get this enabled in production.

**Comment by pmac** [comment]
@alexgibson I've updated it such that the link will show immediately for English locales but hidden with a switch for everything else so that we can enable it only after it's translated broadly enough to enable. Mind taking a quick look at those bits?

**Comment by flodolo** [comment]
I strongly suggest to file a bug to don't forget about removing this conditional.

**Comment by pmac** [comment]
@flodolo would a bug be the best place for the l10n team to update us on the status of this, or should I find it in the l10n tools and keep an eye out myself? If opening a bug with you and Peiying CC'd is the best thing then I'm more than happy to do so.

**Comment by flodolo** [comment]
It should be a bug, since it needs to be addressed by developers. I'd even suggest to assign it to someone, so it ends up in a queue and doesn't get forgotten. Feel free to ping any of us in a couple of weeks to check the status.

**Pull request merged**